### ISSUE TITLE
Why nested templates are bad under Aurelia?

### ISSUE THREAD

**Issue by zakjan** [MeirionHughes/aurelia-template-lint#16]
The readme example says that nested templates are bad under Aurelia. Why so? I'm using nested templates for containerless conditions and cycles. For example:

```
            <template if.bind="!isLoading">
                <template if.bind="isLoadingSucceeded">
                    <template if.bind="models.length > 0">
                        <template repeat.for="model of models">
                            <entity-row model.bind="model"></entity-row>
                        </template>
                    </template>
                    <template if.bind="models.length === 0">
                        <p>No records</p>
                    </template>
                </template>
                <template if.bind="isLoadingFailed">
                    <p>Loading failed</p>
                </template>
            </template>
            <template if.bind="isLoading">
                <p><spinner></spinner></p>
            </template>
```

Can you suggest a better solution? I'd rather not to use nested divs, it could break my design. Also note that it is not possible to use if.bind and repeat.for on the same element. https://github.com/aurelia/binding/issues/250

**Comment by atsu85** [comment]
>  note that it is not possible to use if.bind and repeat.for on the same element

I've listed this idea in aurelia tools linter [issue](https://github.com/aurelia/tools/issues/31) (couple of months before the first commit in this project ;))

**Comment by MeirionHughes** [comment]
hmm I actually don't know if that is valid or not; if it actually works then I'll remove the rule. I was going off the[ aurelia cheat-sheet](http://aurelia.io/docs.html#/aurelia/framework/latest/doc/article/cheat-sheet) which specifically says that: 

```html
<template>
  <table>
    <template repeat.for="customer of customers">
      <tr>
        <td>${customer.fullName}</td>
      </tr>
    </template>
  </table>
</template>
```
is _"Illegal Table Code"._  perhaps @EisenbergEffect can clarify?

It only occurred to me that this has misled me completely. If nested templates work like that I will patch the rule out, or at least make its use optional (config flag to disable it)

For example in latest blog, it talks about using slots for content projection; I think there was only one use-case left for nested templates where you need to use the "replace-part" attribute.

**Comment by zakjan** [comment]
I believe it is illegal only in tables, and it would be great if it worked See https://github.com/aurelia/binding/issues/112

**Comment by MeirionHughes** [comment]
ah ok... I will rewrite it so its only illegal under containers then.

**Comment by zakjan** [comment]
Great, thanks!

**Comment by MeirionHughes** [comment]
Should be able to upgrade to 0.5 and it will stop complaining.
